.github/workflows/bk-ci.yml

```diff
@@ -483,6 +483,14 @@ jobs:
         if: cancelled()
         run: ./dev/ci-tool print_thread_dumps
 
+  typo-check:
+    name: Typo Check
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Check typos
+        uses: crate-ci/typos@master
```

Comment:
The check result shows in the CI log, correct? When we create a new PR and this check triggered, it only checks the newly changed lines or the whole project? 
Due to some public API typo can't be correct, if it checks the whole project all the time, this CI will always fail.

Reply:
@hangc0276 

> The check result shows in the CI log, correct? 

Yes.

> When we create a new PR and this check triggered, it only checks the newly changed lines or the whole project?

It will checks whole project, but it run blazing fast(7seconds). So it wouldn't be a problem.

> Due to some public API typo can't be correct, if it checks the whole project all the time, this CI will always fail.

There is `.typos.toml` to suprression. I have added public API see,

```
# keep for comptability
"deleteable" = "deleteable"
"infinit" = "infinit"
"explict" = "explict"
"uninitalize" = "uninitalize"
```

Reply:
Could you explain the meaning of the suprression?

Reply:
suppression like FP for owasp. I have added our public typos in to suppression list, so it will not fail in CI.